devito/passes/clusters/aliases.py

```diff
@@ -97,22 +107,29 @@ def cire(cluster, mode, sregistry, options, platform):
             break
 
         # Search aliasing expressions
-        aliases = collect(extracted, min_storage, ignore_collected)
+        aliases = collect(extracted, ignore_collected, options)
 
         # Rule out aliasing expressions with a bad flops/memory trade-off
         chosen, others = choose(exprs, aliases, selector)
         if not chosen:
             # Do not waste time
             continue
 
-        # Create Aliases and assign them to Clusters
-        clusters, subs = process(cluster, chosen, aliases, max_par, sregistry, platform)
+        # AliasMapper -> Schedule -> [Clusters]
+        schedule = make_schedule(cluster, aliases, in_writeto, options)
+        schedule = optimize_schedule(cluster, schedule, platform, sregistry, options)
+        clusters, subs = lower_schedule(cluster, schedule, chosen, sregistry, options)
+
+        # The [Clusters] must be ordered so as to reuse as many of the `cluster`'s
+        # IterationIntervals as possible in order to honor the write-to region. This
+        # also guarantees that fusion is maximum
+        processed.extend(clusters)
+        processed.sort(key=partial(cit, cluster))
 
-        # Rebuild `cluster` so as to use the newly created Aliases
-        rebuilt = rebuild(cluster, others, aliases, subs)
+        # Rebuild `cluster` so as to use the newly created aliases
+        rebuilt = rebuild(cluster, others, schedule, subs)
```

Comment:
Isn;t it better to use a cluster-like name rather than past tense of rebuild...?
Forgive if I forgot the art of immutability but can't we just use `cluster = rebuild(cluster,......)`

Reply:
yes, I think we can use `cluster = ...` . Fixing